test/e2e/specs/editor/blocks/links.spec.js

```diff
@@ -8,6 +8,517 @@ test.describe( 'Links', () => {
+				},
+			},
+		] );
+	} );
+
+	test( `is not created when we click away from the link input`, async ( {
+		page,
+		editor,
+		pageUtils,
+	} ) => {
+		// Create a block with some text.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+		} );
+		await page.keyboard.type( 'This is Gutenberg' );
+
+		// Select some text.
+		await pageUtils.pressKeys( 'shiftAlt+ArrowLeft' );
+
+		// Click on the Link button.
+		await page.getByRole( 'button', { name: 'Link' } ).click();
+
+		// Type a URL.
+		await page.keyboard.type( 'https://wordpress.org/gutenberg' );
+
+		// Click somewhere else - it doesn't really matter where.
+		await editor.canvas
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.focus();
+
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: {
+					content: 'This is Gutenberg',
+				},
+			},
+		] );
+	} );
+
+	test( `can be edited`, async ( { page, editor, pageUtils, LinkUtils } ) => {
+		await LinkUtils.createAndReselectLink();
+
+		// Click on the Edit button.
+		await page.getByRole( 'button', { name: 'Edit' } ).click();
+
+		// Change the URL.
+		await page.getByPlaceholder( 'Search or type url' ).fill( '' );
+		await page.keyboard.type( '/handbook' );
+
+		// Submit the link.
+		await pageUtils.pressKeys( 'Enter' );
+
+		// The link should have been updated.
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: {
+					content: 'This is <a href="/handbook">Gutenberg</a>',
+				},
+			},
+		] );
+	} );
+
+	test( `can be removed`, async ( { page, editor, LinkUtils } ) => {
+		await LinkUtils.createAndReselectLink();
+
+		// Unlick via shortcut
+		// we do this to avoid an layout edge case whereby
+		// the rich link preview popover will obscure the block toolbar
+		// under very specific circumstances and screensizes.
+		await page.getByRole( 'button', { name: 'Unlink' } ).nth( 1 ).click();
+
+		// The link should have been removed.
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: {
+					content: 'This is Gutenberg',
+				},
+			},
+		] );
+	} );
+
+	test( `allows Left to be pressed during creation when the toolbar is fixed to top`, async ( {
+		page,
+		editor,
+		pageUtils,
+		LinkUtils,
+	} ) => {
+		await LinkUtils.toggleFixedToolbar( true );
+
+		// Create a block with some text.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+		} );
+		await page.keyboard.type( 'Text' );
+		await page.getByRole( 'button', { name: 'Link' } ).click();
+
+		// Typing "left" should not close the dialog.
+		await pageUtils.pressKeys( 'ArrowLeft' );
+		let popover = page
+			//TODO: change to a better selector when https://github.com/WordPress/gutenberg/issues/51060 is resolved.
+			.locator(
+				'.components-popover__content .block-editor-link-control'
+			);
```

Comment:
@alexstine I wonder if we could solicit your advice here? Currently there is no accessible role for the Link Control UI. We have the format elements but should the control inside the popover have some affordances?

Any guidance on what would improve perceivability for users of assistive technology for this control would be greatly appreciated 🙇

Reply:
yes please, https://github.com/WordPress/gutenberg/issues/51060 🙏

Reply:
Dropped a comment on the issue. Does that help?

Reply:
This will be addressed in https://github.com/WordPress/gutenberg/pull/54063